core/matrix/ell.cpp

```diff
@@ -54,6 +55,18 @@ GKO_REGISTER_OPERATION(outplace_absolute_array,
 }  // namespace ell
 
 
+template <typename ValueType, typename IndexType>
+void Ell<ValueType, IndexType>::validate_data() const
+{
+    GKO_VALIDATE(validation::is_finite(values_),
+                 "matrix must contain only finite values");
+
+    GKO_VALIDATE(
+        validation::ell_is_within_bounds(col_idxs_, 0, this->get_size()[1]),
+        "col_idxs must be within bounds");
+}
```

Comment:
I just noticed: This function doesn't yet check that col_idxs_ doesn't contain duplicate indices, or has indices that appear after padding. We can take a closer look at what this means together, but in general, what you need to do is for each row look at the values `col_idxs_[row + i * this->get_stride()]` for `i = 0, ..., this->get_num_stored_elements_per_row() - 1`. Find the first `invalid_index<IndexType>()` value (if it exists), and all values before that should be unique in that row, all values afterwards should also be `invalid_index<IndexType>()`.